connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectAddArtifactsHandler.scala

```diff
@@ -89,6 +92,28 @@ class SparkConnectAddArtifactsHandler(val responseObserver: StreamObserver[AddAr
     artifactManager.addArtifact(holder, artifact.path, artifact.stagedPath, artifact.fragment)
   }
 
+  protected def uploadStagedArtifactToFS(artifact: StagedArtifact): Unit = {
+    try {
+      val hadoopConf = holder.session.sparkContext.hadoopConfiguration
+      assert(artifact.path.startsWith(forwardToFSPrefix))
+      val destFSPath = new FSPath(
+        Paths.get("/").resolve(artifact.path.subpath(1, artifact.path.getNameCount)).toString
+      )
+      val localPath = artifact.stagedPath
+      val fs = destFSPath.getFileSystem(hadoopConf)
+      fs.copyFromLocalFile(
+        false,
+        true,
+        new FSPath(localPath.toString),
+        destFSPath
+      )
+    } catch {
+      case e: java.io.IOException =>
+        // TODO: report failure in response message
```

Comment:
Discussion: Shall we add a field in ArtifactResponse message for holding failure information here ?

Reply:
I believe @allisonwang-db is working on that.

Reply:
I think we should catch all gRPC errors using this handleError: https://github.com/apache/spark/blob/7d87fecda7077714601cdf7328a11d2719a99bb1/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectService.scala#L119-L123

Reply:
@allisonwang-db 

`handleError` is defined in `SparkConnectService` and it is private, but my code is in `SparkConnectAddArtifactsHandler` class, how can I call `handleError` from `SparkConnectAddArtifactsHandler` code ? Could you make `handleError` to be a utility function that is accessible form other classes ?

Reply:
`SparkConnectAddArtifactsHandler` already defined `onError` handler, see

https://github.com/apache/spark/blob/e8f87413bdd905e5e208ac8484b3b11de1623fc2/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectAddArtifactsHandler.scala#L83

We might need to do some refactor to make it use `handleError` , @allisonwang-db would you mind to do it in follow-up PR ?

Reply:
@MaxGekk will be working on a new framework for error handling. @WeichenXu123 You can migrate to use the new framework when it's completed.